packages/core/admin/utils/create-cache-dir.js

```diff
@@ -115,6 +121,44 @@ async function createCacheDir({ appDir, plugins, useTypeScript }) {
 
   // create plugins.js with plugins requires
   await createPluginsJs(pluginsWithFront, cacheDir);
+
+  // create the tsconfig.json file so we can develop plugins in ts while being in a JS project
+  if (!useTypeScript) {
+    await createTSConfigFile(cacheDir);
+  }
+}
+
+async function createTSConfigFile(dest) {
+  const tsConfig = {
+    compilerOptions: {
+      lib: ['es2019', 'es2020.promise', 'es2020.bigint', 'es2020.string', 'DOM'],
+      noImplicitAny: false,
+      module: 'es2020',
+      target: 'es5',
+      jsx: 'react',
+      allowJs: true,
+      strict: true,
+      moduleResolution: 'node',
+      skipLibCheck: true,
+      esModuleInterop: true,
+      allowSyntheticDefaultImports: true,
+      resolveJsonModule: true,
+      noEmit: false,
+      incremental: true,
+    },
+    include: ['../../../src/admin/src/*', '../../../src/**/**/admin/src/*'],
+    exclude: ['node_modules', '**/*.test.js', '*.js'],
```

Comment:
We could extend the one from @strapi/typescript-utils + update the include/exclude, wdyt?